### PR TITLE
Add strict flag to the JSON parser

### PR DESCRIPTION
**Pull request by eyurtsev**
This updates the default configuration since I think it's almost always what we
want to happen. But we should evaluate whether there are any issues.

### CODE REVIEW

**Comment by eyurtsev** [comment]
see
https://github.com/langchain-ai/langchain/issues/9460

File: libs/langchain/langchain/output_parsers/openai_functions.py [hunk 1]
```diff
@@ -37,11 +37,19 @@ def parse_result(self, result: List[Generation]) -> Any:
 class JsonOutputFunctionsParser(OutputFunctionsParser):
     """Parse an output as the Json object."""
 
+    strict: bool = True
```
Comment (baskaryan) [inline libs/langchain/langchain/output_parsers/openai_functions.py:40]: should we default to false?

Comment (eyurtsev) [reply libs/langchain/langchain/output_parsers/openai_functions.py:40]: Yes I think we should forgot to update

**Review by baskaryan — APPROVED** [review]
lgtm

**Pull request merged**